app/validation/ValidatedActions.scala

```diff
@@ -4,16 +4,30 @@ import auth.TokenSecurity
 import configuration.GraphQLConfiguration
 import controllers.routes
 import graphql.codegen.IsTransferAgreementComplete.{isTransferAgreementComplete => itac}
+import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, Request, Result}
+import services.GetConsignmentService
 
 import scala.concurrent.ExecutionContext
 
-abstract class ValidatedActions() extends TokenSecurity {
+abstract class ValidatedActions() extends TokenSecurity with I18nSupport {
   implicit val ec: ExecutionContext
   val graphqlConfiguration: GraphQLConfiguration
 
   private val isTransferAgreementCompleteClient = graphqlConfiguration.getClient[itac.Data, itac.Variables]()
 
+  def consignmentExists(consignmentId: Long)(f: Request[AnyContent] => Result): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    val getConsignmentService = new GetConsignmentService(graphqlConfiguration)
+    val consignmentExists = getConsignmentService.consignmentExists(consignmentId, request.token.bearerAccessToken)
+    consignmentExists.map {
+      case true => f(request)
+      case false => NotFound(views.html.notFoundError("The consignment you are trying to access does not exist"))
+    }.recover {
```

Comment:
Do we need this recover case?

Presumably, if an error happens here, it's not because of anything the user did? So shouldn't we show an internal server error rather than a bad request? So we should just let the exception happen, since that will generate a 500 page. There's a card in the backlog about adding general exception handling to Play, so that exceptions get converted to the styled 500 page.

Reply:
It's a fair point, I've taken it out